app/src/forms/form/service.js

```diff
@@ -30,16 +30,23 @@ const Rolenames = [Roles.OWNER, Roles.TEAM_MANAGER, Roles.FORM_DESIGNER, Roles.S
 const service = {
   // Get the list of file IDs from the submission
   _findFileIds: (schema, data) => {
-    return (
-      schema.components
-        // Get the file controls
-        .filter((x) => x.type === 'simplefile')
-        // for the file controls, get their respective data element (skip if it's not in data)
-        // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap#for_adding_and_removing_items_during_a_map
-        .flatMap((x) => (data.submission.data[x.key] ? data.submission.data[x.key] : []))
-        // get the id from the data
-        .map((x) => x.data.id)
-    );
+    const findFiles = (components) => {
```

Comment:
Curious about if you have any ideas how we can test this sort of thing. Not that the tests are needed now, because we don't really have a good plan / philosophy about how we want to do testing, but something to think about! We're not able to test everything that someone could do with a form, but I wonder what the "good enough" approach will be!

Reply:
I agree that we should have a testing plan, I can take on this task if you want. In this case though the function is being tested pretty well in app/tests/unit/forms/form/service.spec.js.

Reply:
No need to worry about it now - we'll get there eventually, and we'll probably be focused on frontend tests for awhile.